compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp

```diff
@@ -1500,44 +1500,37 @@ static LogicalResult setRootConfig(mlir::FunctionOpInterface entryPointFn,
 }
 
 /// Sets the lowering configuration for dispatch region for
-/// linalg_ext.winograd.input_transform root op.
+/// the three linalg_ext.winograd ops:
+/// linalg_ext.winograd.filter_transform
+/// linalg_ext.winograd.input_transform
+/// linalg_ext.winograd.output_transform
+/// The vector tile sizes should be 1 for each dim here, because
+/// the winograd decomposition relies on these unit dimensions.
+template <typename WinogradOp>
 static LogicalResult
-setRootConfig(mlir::FunctionOpInterface entryPointFn,
-              IREE::LinalgExt::WinogradInputTransformOp inputOp) {
-  assert(!getLoweringConfig(inputOp) && "expected lowering_config is not set");
-  auto iterationRank = inputOp.getIterationDomainRank();
-  SmallVector<int64_t> vecSizeHints(iterationRank, 1);
-  DistributionHeuristicConfig distConfig;
-  distConfig.vectorSizeHints = vecSizeHints;
-  SmallVector<int64_t> distTileSizes =
-      getDefaultDistributedLevelTileSizes(inputOp, distConfig);
-  TileSizesListType tileSizes;
-  tileSizes.push_back(distTileSizes);
-  SmallVector<int64_t> vecTileSizes(iterationRank, 1);
-  tileSizes.push_back(vecTileSizes);
-  return setOpConfigAndEntryPointFnTranslation(
-      entryPointFn, inputOp, tileSizes,
-      DispatchLoweringPassPipeline::CPULinalgExtTileAndVectorize);
-}
-
-/// Sets the lowering configuration for dispatch region for
-/// linalg_ext.winograd.input_transform root op.
-static LogicalResult
-setRootConfig(mlir::FunctionOpInterface entryPointFn,
-              IREE::LinalgExt::WinogradOutputTransformOp outputOp) {
-  assert(!getLoweringConfig(outputOp) && "expected lowering_config is not set");
-  auto iterationRank = outputOp.getIterationDomainRank();
+setWinogradRootConfig(mlir::FunctionOpInterface entryPointFn,
```

Comment:
I was wondering, instead of templating, can we have `IREE::LinalgExt::LinalgExtOp op` and then do `isa<IREE::LinalgExt::Wino...>(op)'?

Reply:
I think the compilation time check is better. Otherwise, you need to run the program with the winograd input to capture the failure.

Reply:
Yes, I also think the compilation time check is a bit better here.